File: 414-third-maximum-number/414-third-maximum-number.py
class Solution:
    def thirdMax(self, nums: List[int]) -> int:
        nums=set(nums)
        nums=list(nums)
        
        if(len(nums)==1):
            return nums[0]
        if(len(nums)==2):
            return max(nums[0],nums[1])
        
        maxi1=max(nums)
        nums.remove(maxi1)
        maxi2=max(nums)
        nums.remove(maxi2)
        maxi3=max(nums)
        
        return maxi3
            
            

File: 414-third-maximum-number/414-third-maximum-number.cpp
class Solution {
public:
    int thirdMax(vector<int>& nums) {
        
        
        int m=0;
        int ans;
        set<int>  s;
        vector<int> v;
        for(int i=0;i<nums.size();i++)
        {
            s.insert(nums[i]);
        }
        if(s.size()<=2)
        {
            return *max_element(nums.begin(),nums.end());
        }
        int siz=s.size();
        for(auto i: s)
        {
            v.push_back(i);
        }
        
        
        return v[siz-3];
        
        
        
        
        
    }
};